src/components/DataCollector/index.js

```diff
@@ -11,6 +11,25 @@ the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTA
 OF ANY KIND, either express or implied. See the License for the specific language
 governing permissions and limitations under the License.
 */
+const getViewName = xdm => {
```

Comment:
This seems like a fun use case for a util that returns object properties safely instead of all the ifs :) 

Like a util with a try/catch that takes a callback?

Reply:
ok, I will extract it as a util